radix-engine/src/engine/call_frame.rs

```diff
@@ -52,20 +52,37 @@ pub struct CallFrame<
     /// Owned Values
     buckets: HashMap<BucketId, Bucket>,
     proofs: HashMap<ProofId, Proof>,
-    owned_values: ComponentObjects,
+    owned_values: HashMap<StoredValueId, StoredValue>,
 
     /// Referenced values
     worktop: Option<Worktop>,
     auth_zone: Option<AuthZone>,
     component_state: Option<&'p mut ComponentState>,
-    ref_values: HashSet<StoredValueId>,
+    refed_values: HashMap<StoredValueId, ValueRefType>,
 
     /// Caller's auth zone
     caller_auth_zone: Option<&'p AuthZone>,
 
     phantom: PhantomData<I>,
 }
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum ValueType {
+    Owned,
+    Ref(ValueRefType),
+}
```

Comment:
It's strange that we have two hashmaps of values after introducing this enum.

Reply:
Which two hashmaps?

Reply:
Ah, looks like I misread  `owned_values` and `refed_values`. Never mind.